pkg/updater/jobparser.go

```diff
@@ -0,0 +1,337 @@
+/* Copyright (c) 2016 PaddlePaddle Authors All Rights Reserved.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+	 limitations under the License. */
+
+package updater
+
+import (
+	"errors"
+	"fmt"
+	"strconv"
+
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	v1beta1 "k8s.io/api/extensions/v1beta1"
+	apiresource "k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	paddlev1 "github.com/paddlepaddle/edl/pkg/apis/paddlepaddle/v1"
+)
+
+const (
+	imagePullPolicy = "Always"
+)
+
+// JobParser is a interface can parse given simple TrainingJob struct to
+// an integrated TrainingJob
+type JobParser interface {
```

Comment:
The "parser" name is a little confusing, typically parser parses some raw data into a data structure. But judging from the methods, no raw data is involved.

Maybe I have this confusion because `ParseToTrainingJob` takes a `*paddlev1.TrainingJob` and return a `*paddlev1.TrainingJob`, should we make it to take a `job.Spec` rather than `job`?

Maybe we can combine `Validate(job *paddlev1.TrainingJob) error` and `ParseToTrainingJob(job *paddlev1.TrainingJob) *paddlev1.TrainingJob` into a package level function `NewTrainingJob(*Spec) (*paddlev1.TrainingJob, error)`.